### ISSUE TITLE
file_name_format: Unix?

### ISSUE THREAD

**Issue by INPoppoRTUNE** [lukelalo/gallery-card#5]
My files come in the following format: `1721398175-1721398267.mp4`, where the two numbers are Unix formatted dates and times, which seems not to be supported by the card.

I can cut the “start” date using `file_name_date_begins: 11`, but I would need a “Unix” option implemented. I noticed in the docs some datetime difference functionality already implemented with the `AGO` function, so it seems like the necessary libraries are already in place.

Thanks for your time and the beautiful card.

**Comment by lukelalo** [comment]
You need to configure a file sensor, like [this](https://github.com/TarheelGrad1998/files)

**Comment by INPoppoRTUNE** [comment]
> You need to configure a file sensor, like [this](https://github.com/TarheelGrad1998/files)

I may be missing something, but in the attached Git I can't find no reference about the Unix to Human readable format conversion, only sorting(?)

**Comment by lukelalo** [comment]
Yes, you are right. But currently I don’t have enough time to develop this feature. Feel free to do a pull request if you know how to develop it. 
Regards

**Comment by INPoppoRTUNE** [comment]
> Yes, you are right. But currently I don’t have enough time to develop this feature. Feel free to do a pull request if you know how to develop it. Regards

Thanks anyway.
I'm not really familiar with JS, but I'll try to give it a spin.
I'd guess the entire datetime conversion is handled inside of the `_createFileResource(fileRawUrl, fileNameFormat, fileNameDateBegins, captionFormat)`  function in [gallery-card.js](https://github.com/lukelalo/gallery-card/blob/master/src/gallery-card.js), is this correct?

**Comment by lukelalo** [comment]
Yes, that's the right place

**Comment by INPoppoRTUNE** [comment]
> Yes, that's the right place

Here's a snippet that convert Unix timestamps to `YYYYMMDD-HHmmss` (as in [your example](https://github.com/lukelalo/gallery-card#installation)) through external function:
```javascript
function processUnixFileName(fileName) {
  const unixPattern = /^[0-9]+-[0-9]+\.mp4$/;
  if (unixPattern.test(fileName)) {
    const timestamp = fileName.split("-")[0];
    const date = new Date(parseInt(timestamp, 10) * 1000); // Convert Unix timestamp to Date object
    const formattedDate = formatDateToYYYYMMDDHHmmss(date);
    return formattedDate;
  } else {
    return fileName; // If not Unix format, return the original fileName
  }
}

function formatDateToYYYYMMDDHHmmss(date) {
  const yyyy = date.getFullYear();
  const mm = String(date.getMonth() + 1).padStart(2, '0');
  const dd = String(date.getDate()).padStart(2, '0');
  const hh = String(date.getHours()).padStart(2, '0');
  const min = String(date.getMinutes()).padStart(2, '0');
  const ss = String(date.getSeconds()).padStart(2, '0');

  return `${yyyy}${mm}${dd}-${hh}${min}${ss}`;
}

// Example usage
const fileName = "1721398175-1721398267.mp4";
const result = processUnixFileName(fileName);
console.log(result); // Outputs in the YYYYMMDD-HHmmss format: 20240719-140935
```
If I'm correctly reading your code it should be sufficient to add those two functions and call `processUnixFileName(fileName)` inside `_createFileResource(fileRawUrl, fileNameFormat, fileNameDateBegins, captionFormat)`, as it would run only if the Regex pattern for Unix is detected.

**Comment by tomtpcz** [comment]
![Image](https://github.com/user-attachments/assets/42302a33-3aab-4eb7-bd21-000c2123ac1e)
Date conversion would be a great feature :-)

**Comment by INPoppoRTUNE** [comment]
> Date conversion would be a great feature :-)

Check [this answer](https://github.com/JurajNyiri/HomeAssistant-Tapo-Control/discussions/642#discussioncomment-11641742).
